### ISSUE TITLE
(Possible) Layered Navigation Performance Improvement

### ISSUE THREAD

**Issue by rvelhote** [OpenMage/magento-lts#934]
When building the layered navigation tree Magento will [fetch the full attribute option list](https://github.com/OpenMage/magento-lts/blob/2056b4bc8d7f3e033b7996acfc7c9eff24cfc30e/app/code/core/Mage/Eav/Model/Entity/Attribute/Source/Table.php#L54) of all attributes that have _Use In Layered Navigation_ set to _Filterable_.

It might be possible to cache these results using the built-in `initCache` method for collections. I defined two tags that will clear this cache but in order to have the best exprience we should also clear the tag cache for the attribute when saving the attribute  in `Mage_Adminhtml_Catalog_Product_AttributeController`.

```
        if (!isset($this->_options[$storeId])) {
            $key = 'ATTRIBUTE_OPTIONS_ID_' . $this->getAttribute()->getId();
            $tag = array(
                'eav', Mage_Eav_Model_Entity_Attribute::CACHE_TAG,
            );
            $collection = Mage::getResourceModel('eav/entity_attribute_option_collection')
                ->setPositionOrder('asc')
                ->setAttributeFilter($this->getAttribute()->getId())
                ->setStoreFilter($this->getAttribute()->getStoreId())
                ->initCache(Mage::app()->getCache(), $key, $tag)
                ->load();
            $this->_options[$storeId]        = $collection->toOptionArray();
            $this->_optionsDefault[$storeId] = $collection->toOptionArray('default_value');
        }
```
With this change I was able to reduce the number of executed SQL queries by 14 - it will depend on the amount of attributes you have defined of course.

I am happy with this change and I am currently testing in development but I was wondering is anyone sees any drawback/side-effects to this.

Thank you,
Ricardo

**Comment by colinmollenhour** [comment]
Looks like a nice improvement! I don't see any issues with it. Would you submit a PR?

**Comment by rvelhote** [comment]
Thank you for the feedback Colin. I will work on a full PR for further analysis as soon as possible!

**Comment by joshua-bn** [comment]
Should this have been closed with #942 ?

**Comment by sreichel** [comment]
@joshua-bn Thanks.

Fixed with #942.
